File: src/main/java/org/taktik/icure/dao/replicator/ObserverStarter.java
package org.taktik.icure.dao.replicator;

import com.hazelcast.spring.context.SpringAware;
import org.springframework.beans.factory.annotation.Autowired;

import java.io.Serializable;

@SpringAware
public class ObserverStarter implements Runnable, Serializable {
	private transient NewGroupObserver newGroupObserver;

	@Override
	public void run() {
		if (newGroupObserver != null) { newGroupObserver.ensureObserverStarted(); }
	}

	@Autowired
	public void setNewGroupObserver(NewGroupObserver newGroupObserver) {
		this.newGroupObserver = newGroupObserver;
	}
}

File: src/main/kotlin/org/taktik/icure/dao/replicator/ObserverStarter.kt
package org.taktik.icure.dao.replicator

import com.hazelcast.spring.context.SpringAware
import org.springframework.beans.factory.annotation.Autowired

import java.io.Serializable

@SpringAware
class ObserverStarter : Runnable, Serializable {
    @Transient
    private var newGroupObserver: NewGroupObserver? = null

    override fun run() {
        newGroupObserver?.ensureObserverStarted()
    }

    @Autowired
    fun setNewGroupObserver(newGroupObserver: NewGroupObserver) {
        this.newGroupObserver = newGroupObserver
    }
}
